README.rst

```diff
@@ -1,3 +1,55 @@
+Warning
+=======
+
+**New usages of eventlet are now heavily discouraged! Please read the
+following.**
+
+Eventlet was created almost 18 years ago, at a time where async
+features were absent from the CPython stdlib. With time eventlet evolved and
+CPython too, but since several years the maintenance activity of eventlet
+dicreased leading to a growing gap between eventlet and the CPython
+implementation.
+
+This gap is now too high and can lead you to unexpected side effects and bugs
+in your applications.
+
+**Eventlet is now switching to legacy mode**. **Only maintenance for stability
+and bug fixing** will be provided. **No new features will be accepted**, except
+those related to the asyncio hub. **Usage in new projects are discouraged**.
+**Our goal is to plan the retirement of eventlet** and to give you ways to move
+away from eventlet.
+
+If you are looking for a library to manage async network programming,
+and if you do not yet use eventlet, then, we encourage you to use `asyncio`_,
+which is the official async library of the CPython stdlib.
+
+If you already use eventlet, a new asyncio hub has been recently added to
```

Comment:
I would omit this paragraph for now, since:

1. This hub is still very minimally tested. I would like e.g. a few OpenStack projects to run their test suites, at an absolute minimum.
2. Merely using the hub doesn't help with migration, there's a bunch of additional infrastructure (in progress) and documentation needed.
3. Some users can't realistically migrate this way. In particular there's projects like Celery that specifically rely on having an API that looks blocking, so "rewrite with asyncio" doesn't meet that use case. Might be a bad desire, but it's still a different case than OpenStack. So I suspect we'll want different advice for different users ("you can use gevent, although we believe it will have similar issues" for celery-like users I guess).

Reply:
Good points, let me drop that sentence.

Reply:
What do you think if we reword this sentence to simply indicate that we are thinking about a migration plan?
Maybe would should less formal about the existence of this  feature, but we could simply indicate that we are looking for a solution to allow user migrating from eventlet to asyncio without much details for now, thoughts?